attribute.go

```diff
@@ -24,6 +25,29 @@ type Attribute struct {
 	Data []byte
 }
 
+// IsNested interprets the Type field of the Attribute to determine whether it's
+// a nested attribute or not. This facilitates recursive parsing of the attribute.
+// This is the leftmost bit in Type. Mutually exclusive with IsNetByteOrder().
+func (a Attribute) IsNested() bool {
+	return (a.Type & unix.NLA_F_NESTED) > 0
+}
+
+// IsNetByteOrder interprets the Type field of the Attribute to determine whether the attribute
```

Comment:
@mdlayher Your initial guess was correct, this flag doesn't indicate a change to the endianness of the header itself. It only seems to be used in IPSet. Following [an example call chain `nla_put_net16`](http://lxr.free-electrons.com/source/net/netfilter/ipset/ip_set_hash_ipport.c#L69) all the way down to [`__nla_put`](http://lxr.free-electrons.com/source/lib/nlattr.c#L482):

```
 void __nla_put(struct sk_buff *skb, int attrtype, int attrlen,
                              const void *data)
{
        struct nlattr *nla;

        nla = __nla_reserve(skb, attrtype, attrlen);
        memcpy(nla_data(nla), data, attrlen);
}
```

In the example above, `attrtype` is `IPSET_ATTR_PORT` OR'ed with `NLA_F_NET_BYTEORDER`. `data` is a `(struct hash_ipport4_elem)->data`, which is supposed to be a big-endian representation of a 16-bit port number. `attrtype` has the big-endian bit enabled to indicate this. Supporting this bit would allow us to successfully interpret IPSet in a NetFilter package. 👍